### PR TITLE
ascanrulesAlpha: add rule for MongoDB time tests

### PR DESCRIPTION
**Pull request by thc202**
Move time based tests to its own rule, ID 90039.

Part of zaproxy/zaproxy#7341.

---
Extracted from #4316.

### CODE REVIEW

**Comment by thc202** [comment]
Ready for review.

**Review by kingthorin — APPROVED** [review]
LGTM

File: addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRule.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
```
Comment (psiinon) [inline addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRule.java:6]: 2024?

File: addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRule.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                    "0)  || sleep({0})",
+                    // Attacking mapReduce clause (only for old versions of mongodb)
+                    "_id);}, function(inj) { sleep({0});return 1;}, { out: 'x'}); db.injection.mapReduce(function() { emit(1,1");
+
+    private static final Logger LOGGER = LogManager.getLogger(MongoDbInjectionTimingScanRule.class);
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(
+                    CommonAlertTag.OWASP_2021_A03_INJECTION,
+                    CommonAlertTag.OWASP_2017_A01_INJECTION,
+                    CommonAlertTag.WSTG_V42_INPV_05_SQLI);
```
Comment (psiinon) [inline addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRule.java:90]: I think a `CommonAlertTag.TEST_TIMING` or similar would be _really_ useful.
Esp if/when (hopefully when) we support tag based policy rules, eg "inc: ["OWASP_2021_*"], exc: ["TEST_TIMING"]` if you see what I mean...

File: addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRuleUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
```
Comment (psiinon) [inline addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRuleUnitTest.java:6]: 2024

**Review by psiinon — CHANGES_REQUESTED** [review]
Looks good, just a couple of suggestions..

**Comment by thc202** [comment]
Comments addressed.

**Review by kingthorin — APPROVED** [review]

**Review by psiinon — APPROVED** [review]

**Pull request merged**